L1Trigger/L1TMuonEndCap/src/PrimitiveConversion.cc

```diff
@@ -495,19 +511,23 @@ void PrimitiveConversion::convert_rpc(
     double glob_phi   = emtf::rad_to_deg(gp.phi().value());
     double glob_theta = emtf::rad_to_deg(gp.theta());
     double glob_eta   = gp.eta();
+    double glob_rho   = gp.perp();
+    double glob_z     = gp.z();
 
     // Use RPC-specific convention in docs/CPPF-EMTF-format_2016_11_01.docx
     // Phi precision is 1/15 degrees (11 bits), 4x larger than CSC precision of 1/60 degrees (13 bits)
     // Theta precision is 36.5/32 degrees (5 bits), 4x larger than CSC precision of 36.5/128 degrees (7 bits)
-    int fph = emtf::calc_phi_loc_int(glob_phi, sector_, 11);
-    int th  = emtf::calc_theta_int(glob_theta, conv_hit.Endcap(), 5);
-
-    if (not(0 <= fph && fph < 1250))
-      { edm::LogError("L1T") << "fph = " << fph; return; }
-    if (not(0 <=  th &&  th < 32))
-      { edm::LogError("L1T") << "th = " << th; return; }
-    if (not(th != 0b11111))  // RPC hit valid when data is not all ones
-      { edm::LogError("L1T") << "th = " << th; return; }
+    //
+    // NOTE: fph and th are recalculated using CPPF LUTs in the convert_rpc_details() function,
+    //       this part is still kept because it is needed for Phase 2 iRPC hits.
+    //
+    int fph = emtf::calc_phi_loc_int_rpc(glob_phi, conv_hit.PC_sector());
+    int th  = emtf::calc_theta_int_rpc(glob_theta, conv_hit.Endcap());
+
+    //assert(0 <= fph && fph < 1024);
+    assert(0 <= fph && fph < 1250);
+    assert(0 <=  th &&  th < 32);
+    assert(th != 0b11111);  // RPC hit valid when data is not all ones
     fph <<= 2;  // upgrade to full CSC precision by adding 2 zeros
```

Comment:
Should this be done in calc_phi_loc_int_rpc and calc_theta_int_rpc instead?

Reply:
The assert statements? I think it's more clear here.

Reply:
No, I mean the bit-shift by 2.  I realize I was the one who did it this way originally (I think), but would it be clearer bit-shift inside calc_phi_loc_int_rpc and calc_theta_int_rpc?  Then these functions would return the final integer scale used by EMTF, rather than an intermediate scale.  Of course the intermediate scale is what CPPF actually transmits ...

Reply:
They can be changed, but I think they will trigger changes in some other places. I'd think it's not worth to change it now, unless there is a strong motivation.